src/daemon/distribution.c

```diff
@@ -0,0 +1,148 @@
+#include <math.h>
+#include "collectd.h"
+#include "distribution.h"
+
+struct bucket_s{
+	size_t bucket_counter;
+	double min_boundary;
+	double max_boundary;
+};
+
+struct distribution_s {
+     bucket_t *buckets;
+     size_t num_buckets;
+     size_t total_scalar_count;
+     double raw_data_sum;
+};
+
+//bucket constructor
+bucket_t initialize_bucket(double min_boundary, double max_boundary) {
```

Comment:
This appears to be a private function. In C, you use the `static` keyword to signal that a function is private:

```c
static bucket_t initialize_bucket(…)
```

Reply:
Okay, thank you.